README.md

```diff
@@ -7,40 +7,89 @@ its current limitations.
 
 # Usage
 
-## installation
+## Installation
 
-This library is intended to be used with maven buildsystem, although you can
-probably easily move it to any other if you're familiar with those. To add it
-to your mvn project edit the pom.xml file to add:
+This library is intended to be used with maven build system, although you can
+probably easily move it to any other if you're familiar with those. To add it to
+your mvn project edit the pom.xml file to add:
 
 ```xml
     ...
-    <dependency>
-      <groupId>io.github.in-toto</groupId>
-      <artifactId>in-toto</artifactId>
-      <version>0.1</version>
-      <scope>compile</scope>
-    </dependency>
+<dependency>
+  <groupId>io.github.in-toto</groupId>
+  <artifactId>in-toto</artifactId>
+  <version>0.3.3</version>
+</dependency>
     ...
 ```
 
-With it you should be able to use the library inside of your project.
+With it, you should be able to use the library inside your project.
 
-## Using the library
+## Using the new library
 
-The library exposes a series of objects and convenience methods to create,
-sign, and serialize in-toto metadata. As of now, only Link metadata is
-supported (see the Limitations section to see what exactly is supported as of
-now).
+The library exposes a new set of models used for in-toto 0.1.0, DSSE 1.0.0 and .
+If you wish to use the deprecated legacy Link library, please skip to the next
+section.
 
-Metadata classes are located in the `io.in_toto.models.*` namespace. You can,
-for example create a link as follows:
+The new library allows you to instantiate a Statement and populate it as
+follows:
 
 ```java
-    Link link = new Link(null, null, "test", null, null);
+Subject subject=new Subject();
+    subject.setName("curl-7.72.0.tar.bz2");
+    subject.setDigest(
+    Map.of(
+    DigestSetAlgorithmType.SHA256.toString(),
+    "d4d5899a3868fbb6ae1856c3e55a32ce35913de3956d1973caccd37bd0174fa2"));
+    Predicate predicate=new Predicate(); // Let's pretend this is an SLSA predicate
+    Statement statement=new Statement();
+    statement.set_type(StatementType.STATEMENT_V_0_1);
+    statement.setSubject(List.of(subject));
+    statement.setPredicateType(PredicateType.SLSA_PROVENANCE_V_0_1);
+    statement.setPredicate(predicate);
 ```
 
-This will create a link object that you can operate with. 
+Finally, you can use the built-in `IntotoHelper` class to validate and transform
+it into its JSON representation as follows:
+
+```java
+    String jsonStatement=IntotoHelper.validateAndTransformToJson(statement);
+```
+
+If the statement passed to the method is malformed the library will throw
+an `InvalidModelException` that will contain a message with the errors.
+
+If you, however wish to create a DSSE based In-toto envelope, The library
+features a convenience method:
+
+```java
+IntotoEnvelope intotoEnvelope=IntotoHelper.produceIntotoEnvelope(statement,signer);
+```
+
+This method will accept a `io.github.intoto.models.Statement` and an
```

Comment:
nit: the method exists now so maybe use present tense instead of past?  E.g. "This method accepts" or maybe just leave it out entirely since folks can go look at the code/docs?